snuba/query/snql/parser.py

```diff
@@ -902,7 +910,7 @@ def mangle_column_value(exp: Expression) -> Expression:
 
         return replace(exp, column_name=f"{alias_prefix}{exp.column_name}")
 
-    query.transform_expressions(mangle_aliases)
+    query.transform_expressions(mangle_aliases, skip_array_join=True)
```

Comment:
Pretty sure we ran into a problem when mangling aliases in array joins @lynnagara  do you remember the context ?

Reply:
That's why this skips the array join clause. You can't use aliases in the array join.

Reply:
How come SnQL needs it's own alias mangling function? Can't it use the same code as https://github.com/getsentry/snuba/blob/master/snuba/query/parser/__init__.py#L396

Reply:
There's a couple reasons: The original function doesn't support sub queries and it also sets the arrayjoin parameter which I don't want. It's pretty simple so I figured copying it would be easier than trying to abstract it.